File: BE/ari/src/main/java/com/ccc/ari/music/mapper/TrackMapper.java
package com.ccc.ari.music.mapper;

import com.ccc.ari.music.domain.track.TrackDto;
import com.ccc.ari.music.domain.track.TrackEntity;

public class TrackMapper {

    public static TrackDto toDto(TrackEntity entity) {
        return TrackDto.builder()
                .trackId(entity.getTrackId())
                .title(entity.getTrackTitle())
                .composer(entity.getComposer())
                .lyricist(entity.getLyricist())
                .lyrics(entity.getLyrics())
                .trackNumber(entity.getTrackNumber())
                .trackFileUrl(entity.getTrackFileUrl())
                .trackLikeCount(entity.getTrackLikeCount())
                // TODO : genre 구현되고 이 부분 수정해야됨. 그리고 AlbumEntity 수정하면서 albumId 부분 수정해야됨.
                .albumId(entity.getAlbum().getAlbumId())
                .gereName(entity.getAlbum().getGenre().getGenreName())
                .build();
    }
}

File: FE/lib/data/mappers/track_mapper.dart
import 'package:ari/data/models/track.dart' as data;
import 'package:ari/domain/entities/track.dart' as domain;

/// data.Track → domain.Track 변환 확장 메서드
extension DataTrackMapper on data.Track {
  domain.Track toDomainTrack() {
    print('🔁 [Mapper] toDomainTrack 실행 - id: $id');

    return domain.Track(
      trackId: id,
      trackTitle: trackTitle,
      artistName: artist,
      albumId: albumId,
      coverUrl: coverUrl,
      lyric: lyrics,
      composer: composer.split(', '), // string → list
      lyricist: lyricist.split(', '),
      trackFileUrl: trackFileUrl,
      trackNumber: 0,
      trackLikeCount: trackLikeCount,
      commentCount: 0,
      comments: [],
      createdAt: '',
      albumTitle: '', // 서버에 없으면 빈 값 처리
      genreName: '', // 서버에 없으면 빈 값 처리
    );
  }
}
